src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/ByteArrayModelBinder.cs

```diff
@@ -11,6 +14,25 @@ namespace Microsoft.AspNetCore.Mvc.ModelBinding.Binders
     /// </summary>
     public class ByteArrayModelBinder : IModelBinder
     {
+        protected readonly ILogger logger;
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="ByteArrayModelBinder"/>.
+        /// </summary>
+        public ByteArrayModelBinder()
+            : this(NullLoggerFactory.Instance)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="ByteArrayModelBinder"/>.
+        /// </summary>
+        /// <param name="loggerFactory">The <see cref="ILoggerFactory"/>.</param>
+        public ByteArrayModelBinder(ILoggerFactory loggerFactory)
+        {
+            logger = loggerFactory.CreateLogger(GetType());
+        }
+
         /// <inheritdoc />
         public Task BindModelAsync(ModelBindingContext bindingContext)
```

Comment:
@dougbu regarding 

> Stopping as I've realized significantly less would need to change if ModelBinderContext exposed a public virtual ILoggerFactory property. Do this in ModelBinderProviderContext instead if either logger creation is fairly expensive or a provider does some logging itself

So you are suggesting to create logger in the BindModelAsync using the LoggerFactory exposed on the binding context, right? Sure it would reduce the code changes to constructors of the binders, but just wondering if its good to do this on every request (I am referring to https://github.com/aspnet/Logging/blob/dev/src/Microsoft.Extensions.Logging/LoggerFactory.cs#L70)

Reply:
I listed a number of options. `ModelBinderProviderContext` might be a better one.

Reply:
Could you hash out what it looks like when done via the calling code \ context? We could look at cleaning up some of the logging code if it starts showing up in the hot path.